orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,265 @@
+            return 0
+
+
+class OWConcordance(OWWidget):
+    name = "Concordance"
+    description = "Display the context of the word."
+    icon = "icons/Concordance.svg"
+    priority = 30000
+
+    inputs = [('Corpus', Table, 'set_corpus')]
+    outputs = [('Selected Documents', Table, )]
+
+    autocommit = Setting(True)
+    width = Setting(5)
+    word = Setting("")
+    # TODO Set selection settings.
+
+    def __init__(self):
+        super().__init__()
+
+        self.corpus = None      # Corpus
+        self.n_documents = ''   # Info on docs
+        self.n_matching = ''    # Info on docs matching the word
+        self.n_tokens = ''      # Info on tokens
+        self.n_types = ''       # Info on types (unique tokens)
+
+        # Info attributes
+        info_box = gui.widgetBox(self.controlArea, 'Info')
+        gui.label(info_box, self, 'Documents: %(n_documents)s')
+        gui.label(info_box, self, 'Tokens: %(n_tokens)s')
+        gui.label(info_box, self, 'Types: %(n_types)s')
+        gui.label(info_box, self, 'Matching: %(n_matching)s')
+
+        # Width parameter
+        gui.spin(self.controlArea, self, 'width', 3, 10, box=True,
+                 label="Number of words:", callback=self.set_width)
+
+        gui.rubber(self.controlArea)
+
+        # Search
+        c_box = gui.widgetBox(self.mainArea, orientation="vertical")
+        self.input = gui.lineEdit(
+            c_box, self, 'word', orientation=Qt.Horizontal,
+            sizePolicy=QSizePolicy(QSizePolicy.MinimumExpanding,
+                                   QSizePolicy.Fixed),
+            label='Query:', callback=self.refresh_search, callbackOnType=True)
+        self.input.setFocus()
+
+        # Concordances view
+        self.conc_view = QTableView()
+        self.model = ConcordanceModel()
+        self.conc_view.setModel(self.model)
+        self.conc_view.setWordWrap(False)
+        self.conc_view.setSelectionBehavior(QTableView.SelectRows)
+        self.conc_view.setSelectionModel(DocumentSelectionModel(self.model))
+        # connect selectionChanged to self.commit(), which will be
+        # updated by gui.auto_commit()
+        self.conc_view.selectionModel().selectionChanged.connect(lambda:
+                                                                 self.commit())
+        self.conc_view.horizontalHeader().hide()
+        self.conc_view.setShowGrid(False)
+        self.mainArea.layout().addWidget(self.conc_view)
+        self.set_width()
+
+        # Auto-commit box
+        gui.auto_commit(self.controlArea, self, 'autocommit', 'Commit',
+                        'Auto commit is on')
+
+    def sizeHint(self): # pragma: no cover
+        return QSize(600, 400)
+
+    def set_width(self):
+        sel = self.conc_view.selectionModel().selection()
+        self.model.set_width(self.width)
+        if sel:
+            self.conc_view.selectionModel().select(sel,
+                QItemSelectionModel.SelectCurrent|QItemSelectionModel.Rows)
```

Comment:
And since you'll be changing the file anyway, you can add a space before and after `|`.